packages/fxa-settings/src/components/ThirdPartyAuth/index.tsx

```diff
@@ -2,20 +2,140 @@
+  const storageData = useStorage();
+  const account = useAccount();
+
+  useEffect(() => {
+    // TODO: Figure out why `storageData` is not available
+    const authParams = Storage.factory('localStorage', window).get(
+      'fxa_third_party_params'
+    );
+    if (authParams) {
+      completeSignIn();
+    }
+  }, []);
+
   if (!enabled) {
     return null;
   }
 
+  /**
+   * signIn - Handles the sign-in process for third-party authentication providers.
+   * Creates a form, sets necessary parameters, and submits it to the provider's
+   * authorization endpoint.
+   *
+   * @param {ThirdPartyAuthSigninParams} config - Configuration parameters for the signIn process.
+   */
+  function signIn(config: ThirdPartyAuthSigninParams) {
+    clearStoredParams();
+
+    // We stash originating location in the state oauth param
+    // because we will need it to use it to log the user into FxA
+    const currentParams = new URLSearchParams(window.location.search);
+    currentParams.delete('deeplink');
+    currentParams.set('showReactApp', 'true');
+
+    const state = encodeURIComponent(
+      `${window.location.origin}${
+        window.location.pathname
+      }?${currentParams.toString()}`
+    );
+
+    const form = window.document.createElement('form');
+
+    form.setAttribute('method', 'GET');
+    form.setAttribute('action', config.authorizationEndpoint);
+
+    const params: Record<string, string | undefined> = {
+      client_id: config.clientId,
+      scope: config.scope,
+      redirect_uri: config.redirectUri,
+      state,
+      access_type: 'offline',
+      prompt: 'consent',
+      response_type: config.responseType,
+      response_mode: config.responseMode,
+    };
+
+    for (const [key, value] of Object.entries(params)) {
+      if (!value) {
+        continue;
+      }
+      const input = window.document.createElement('input');
+      input.setAttribute('type', 'hidden');
+      input.setAttribute('name', key);
+      input.setAttribute('value', value);
+      form.appendChild(input);
+    }
+
+    // To avoid any CORs issues we append the form to the body and submit it.
+    window.document.body.appendChild(form);
+
+    form.submit();
+  }
+
+  async function completeSignIn() {
+    try {
+      const authParams = Storage.factory('localStorage', window).get(
+        'fxa_third_party_params'
+      );
+      const code = authParams.code;
+      const provider = authParams.provider || 'google';
+
+      // Verify and link the third party account to FxA. Note, this
+      // will create a new FxA account if one does not exist.
+      await account.verifyAccountThirdParty(code, provider);
+
+      // TODO: The response from above contains a session token
+      // which can be used to sign the user in to FxA or used
+      // to complete an Oauth flow.
+    } catch (error) {
+      // Fail silently on errors, this could be some leftover
+      // state from a previous attempt.
+      console.log(error);
```

Comment:
Q: Do these get logged to Sentry or just silently to the user's DevTools console?